x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/execution/search/extractor/AbstractFieldHitExtractor.java

```diff
@@ -81,12 +83,62 @@ public Object extract(SearchHit hit) {
         Object value = null;
         DocumentField field = null;
         if (hitName != null) {
-            // a nested field value is grouped under the nested parent name (ie dep.dep_name lives under "dep":[{dep_name:value}])
-            field = hit.field(hitName);
+            value = extractNestedField(hit);
         } else {
             field = hit.field(fieldName);
+            if (field != null) {
+                value = unwrapFieldsMultiValue(field.getValues());
+            }
+        }
+        return value;
+    }
+
+    /*
+     * For a tree of fields like root.nested1.nested2.leaf where nested1 and nested2 are nested field types,
```

Comment:
nit: would "a **path** of fields" make more sense? (since I guess the doc isn't parsed/accessed as a tree?)